akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -230,63 +237,46 @@ private[akka] object StreamLayout {
     override def materializedValueComputation: MaterializedValueNode = Ignore
   }
 
-  final case class CompositeModule(
-    subModules: Set[Module],
-    shape: Shape,
-    override val connections: List[(OutPort, InPort)],
-    override val materializedValueComputation: MaterializedValueNode,
-    attributes: OperationAttributes) extends Module {
+  final case class CopiedModule(shape: Shape, attributes: OperationAttributes, copyOf: Module) extends Module {
+    override val subModules: Set[Module] = Set(copyOf)
+
+    override def withAttributes(attr: OperationAttributes): Module = this.copy(attributes = attr)
+
+    override def carbonCopy: Module = this.copy(shape = shape.deepCopy())
```

Comment:
Well, it’s a trade-off—we could just store the CopiedModule on the stack and adapt `currentLayout` to call `copyOf`—but I think this is okay, it saves some pointer chasing. Please add a comment to that stack that it stores the copies’ originals.